src/services/services.ts

```diff
@@ -5724,6 +5696,23 @@ module ts {
         return { getClassificationsForLine };
     }
 
+    /// getDefaultLibraryFilePath
+    declare var __dirname: string;
+    declare var module: any;
+
+    /**
+      * Get the path of the default library file (lib.d.ts) as distributed with the typescript
+      * node package.
+      * The functionality is not supported if the ts module is consumed outside of a node module. 
+      */
+    export function getDefaultLibraryFilePath(options: CompilerOptions): string {
+        if (typeof module !== "undefined" && module.exports) {

```

Comment:
If you just want to use `__dirname` then check `typeof __dirname`. This current line isn’t a good check for being in a Node.js environment. You can get the same invalid answer with a `define(function (require, exports, module) {})` AMD wrapper or if someone just pollutes some globals. Probably the “best” check for knowing if you are in an environment that is intentionally trying to quack like Node.js is `process && process.versions && process.versions.node` but in any case actually trying to detect the API you want to use is best.